MyOwnORM/MyOwnORM/CustomDbSetReflection.cs

```diff
@@ -0,0 +1,693 @@
+﻿using Microsoft.CSharp.RuntimeBinder;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyOwnORM
+{
+    public class CustomDbSetReflection<T> where T : class
+    {
+        private protected static T MapReaderToEntity(SqlDataReader reader)
```

Comment:
Your classes should have only 1 responsibility(now it is working with reflection, so separate out of here working with sql, queries, expressions)

Reply:
Fixed